### PR TITLE
Manually copy entire Python bindings docs folder instead of just `api.html`

### PR DESCRIPTION
**Pull request by sea-bass**
This furthers the hack we employ to get Python bindings docs working with Sphinx multiversion...

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -149,7 +149,7 @@ jobs:
         path: build/html
 
   deploy:
-    if: github.repository_owner == 'ros-planning' && github.ref == 'refs/heads/main'
+    if: github.repository_owner == 'ros-planning'
```
Comment (eholum) [inline .github/workflows/deploy.yml:152]: Oh is this going to murderize main?

Comment (sea-bass) [reply .github/workflows/deploy.yml:152]: Definitely. I'm just speeding up my testing and will revert this line if this ends up working.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -138,9 +138,9 @@ jobs:
     # TODO(henningkayser): fix hack for using python api artifact in multiversion
     - name: Build multiversion
       run: |
-        cp build/html/main/doc/api/python_api/api.html .   # backup artifact html
+        cp -r build/html/main/doc/api/python_api/ .   # backup artifact html
         make multiversion
-        cp -f api.html build/html/main/doc/api/python_api/ # restore artifact html
+        cp -rf api.html build/html/main/doc/api/      # restore artifact html
```
Comment (eholum) [inline .github/workflows/deploy.yml:143]:
```suggestion
        cp -rf python_api/* build/html/main/doc/api/python_api      # restore artifact html
```

**Review by eholum — COMMENTED** [review]
whoops. might be some directory shenanigans...

**Review by eholum — APPROVED** [review]

**Pull request merged**